typed-racket-lib/typed-racket/rep/prop-rep.rkt

```diff
@@ -143,8 +143,7 @@
   [#:for-each (f) (for-each f ps)]
   [#:custom-constructor/contract
    (-> (listof (or/c TypeProp? NotTypeProp? LeqProp?)) OrProp?)
-   (let ([ps (sort ps (λ (p q) (unsafe-fx<= (eq-hash-code p)
```

Comment:
@jackfirth my guess is that this change is slower

Reply:
If it is, that seems like a missed optimization opportunity in `sort`.